### ISSUE TITLE
MAMA JNI doesn't exit properly when using non-default threads for processing

### ISSUE THREAD

**Issue by fquinner** [OpenMAMA/OpenMAMA#181]
## Summary
You can recreate by running this in the publisher:

```
capturereplayc -S OM -f ../../OpenMAMA-test-data/playbacks/openmama_utpcasheuro_capture.5000.10.qpid.mplay -m qpid -tport pub -dictionary ../../OpenMAMA-test-data/dictionaries/data.dict -v
```

And this on the mamajni side:

```
java -cp ../../OpenMAMA/openmama_install_2.4.0/lib/mamajni.jar com.wombat.mama.examples.MamaListen -S OM -m qpid -tport sub -f ../../OpenMAMA-test-data/playbacks/openmama_utpcasheuro_capture.5000.10.sym -shutdown 30 -threads 1
```
## Areas Affected
- [ ] MAMAC
- [ ] MAMACPP
- [ ] MAMADOTNET
- [X] MAMAJNI
- [ ] MAMDA
- [ ] MAMDACPP
- [ ] MAMDADOTNET
- [ ] MAMDAJNI
- [ ] Visual Studio
- [ ] SCons
- [ ] Unit Tests
- [ ] Examples

## Details
The application hangs when trying to exit, but only when `-threads X` is provided. This requires investigation to determine if the example apps or MAMA JNI is at fault here.

**Comment by reed-jpm-alpert** [comment]
I think this is due to using AttachCurrentThread() rather than AttachCurrentThreadAsDaemon() in the JNI layer when it attaches a subscription callback thread to a JVM thread. It's not really possible to detach one of these threads, since the bridge code that terminates the thread does not know that the language is Java. Making these threads daemon ones will allow the JVM to exit even if they are active.

**Comment by fquinner** [comment]
Implemented in #210
